cmd/internal/find/parser/expression/parser.go

```diff
@@ -141,10 +194,11 @@ func (parser *parser) Parse(tokens []string) (predicate.Predicate, []string, err
 	// := operator's scoping rules. tks is used to avoid accidentally
 	// overwriting tokens.
 	//
-	// POST-LOOP INVARIANT: err == nil or err is an UnknownTokenError
+	// POST-LOOP INVARIANT: err == nil or err is an UnknownTokenError/IncompleteOperatorError
 	var p predicate.Predicate
 	var tks []string
 	var err error
+Loop:
```

Comment:
Could this be a function that we return from instead? There's a lot going on in this function already.

Reply:
Updated the PR. I wasn't sure if it added much clarity at first, but looking at it now, I think that it does.